tick/optim/prox/src/prox_binarsity.cpp

```diff
@@ -0,0 +1,93 @@
+
+
+#include "prox_tv.h"
+#include "prox_binarsity.h"
+
+
+ProxBinarsity::ProxBinarsity(double strength,
+                             SArrayULongPtr blocks_start,
+                             SArrayULongPtr blocks_length,
+                             bool positive)
+        : Prox(strength)
+{
+    this->blocks_start = blocks_start;
```

Comment:
Check that block_start and block_length have the same size, even if this has been checked in Python (always check in C++ as it could lead to a segfault)